### PR TITLE
Initial fix - fixing makefile flags and compiler warnings

### PR DESCRIPTION
**Pull request by RogFanther**
Hello, this is initially to fix the use of old freetype-config to allow compilation in modern Haiku, and also replacement of some unsafe sprintf() calls with BString class.

### CODE REVIEW

File: bepdf/Makefile [hunk 1]
```diff
@@ -153,7 +153,8 @@ DEBUGGER :=
 COMPILER_FLAGS = -Wno-write-strings
 
 #	Specify any additional linker flags to be used.
-LINKER_FLAGS = $(shell freetype-config --libs)
+#LINKER_FLAGS = $(shell freetype-config --libs)
+LINKER_FLAGS = $(shell pkg-config --cflags --libs freetype2)
```
Comment (korli) [inline bepdf/Makefile:157]: Why --cflags?

File: bepdf/beos/FileInfoWindow.cpp [hunk 2]
```diff
@@ -87,7 +87,7 @@ static int ToInt(const char *s, int i, int n) {
 static const char *ToDate(const char *date, time_t *time) {
 static char s[80];
 	struct tm d;
-	memset(&d, sizeof(d), 0);
+    memset(&d,0, sizeof(d));
```
Comment (korli) [inline bepdf/beos/FileInfoWindow.cpp:90]:
```suggestion
    memset(&d, 0, sizeof(d));
```

File: xpdf/Makefile [hunk 3]
```diff
@@ -191,10 +191,12 @@ SYMBOLS :=
...
 
 #	Specify any additional compiler flags to be used.
-COMPILER_FLAGS := $(shell freetype-config --cflags)
+#COMPILER_FLAGS := $(shell freetype-config --cflags)
+COMPILER_FLAGS := $(shell pkg-config --cflags freetype2)
 
 #	Specify any additional linker flags to be used.
-LINKER_FLAGS = $(shell freetype-config --libs)
+#LINKER_FLAGS = $(shell freetype-config --libs)
+LINKER_FLAGS = $(shell pkg-config --cflags freetype2)
```
Comment (korli) [inline xpdf/Makefile:199]: Why not --libs?

File: bepdf/beos/PreferencesWindow.cpp [hunk 4]
```diff
@@ -59,13 +59,14 @@ void PreferencesWindow::UpdateWorkspace() {
 }
 
 void PreferencesWindow::BuildWorkspaceMenu(BMenu *m) {
-	m->AddItem(new BMenuItem(B_TRANSLATE("current"), new BMessage(WORKSPACE_CHANGED)));
+	m->AddItem(new BMenuItem(B_TRANSLATE("Current"), new BMessage(WORKSPACE_CHANGED)));
 	m->AddSeparatorItem();
 	int n = count_workspaces();
+
+    BString buffer;
```
Comment (korli) [inline bepdf/beos/PreferencesWindow.cpp:66]:
```suggestion
	BString buffer;
```

File: bepdf/beos/PreferencesWindow.cpp [hunk 5]
```diff
@@ -59,13 +59,14 @@ void PreferencesWindow::UpdateWorkspace() {
...
 	for (int i = 1; i <= n; i ++) {
-		char buffer[10];
-		sprintf(buffer, "%d", i);
-		m->AddItem(new BMenuItem(buffer, new BMessage(WORKSPACE_CHANGED)));
+        buffer.SetToFormat("%d", i);
```
Comment (korli) [inline bepdf/beos/PreferencesWindow.cpp:68]:
```suggestion
		buffer.SetToFormat("%d", i);
```

File: bepdf/beos/FileInfoWindow.cpp (hunk 2 above)
Comment (RogFanther) [reply bepdf/beos/FileInfoWindow.cpp:90]: The formatting, yes ? I overlooked that, your editing is better.

File: bepdf/beos/PreferencesWindow.cpp (hunk 5 above)
Comment (RogFanther) [reply bepdf/beos/PreferencesWindow.cpp:68]: Oh, I got it. Tabs versus spaces. But shouldn´t this buffer... get aligned with the previous ( m-> ) line ?

File: bepdf/beos/PreferencesWindow.cpp (hunk 4 above)
Comment (RogFanther) [reply bepdf/beos/PreferencesWindow.cpp:66]: Oh, Tabs x Spaces , sorry.

**Comment by RogFanther** [comment]
Hi, 

Should I answer these here or reply to the comments in github ? Sorry, still getting used to this.

In this one I overlooked the space, and the others were those Tabs x Spaces thing. I will make an effort to pay attention to that in the future.

The Makefile editions I will need to find the research I did and test more things. Was trying to understand what was the problem with that freetype-config so tried a lot of things ( and forgot to remove that commented line also ) .

Thanks, 

  ----- Original Message -----

File: bepdf/beos/PreferencesWindow.cpp (hunk 5 above)
Comment (korli) [reply bepdf/beos/PreferencesWindow.cpp:68]: Sure.

**Comment by korli** [comment]
Please fix in your branch, this will update in this pull request. This will reviewed again, and eventually pushed.

File: bepdf/beos/PreferencesWindow.cpp [hunk 6]
```diff
@@ -59,13 +59,14 @@ void PreferencesWindow::UpdateWorkspace() {
 }
 
 void PreferencesWindow::BuildWorkspaceMenu(BMenu *m) {
-	m->AddItem(new BMenuItem(B_TRANSLATE("current"), new BMessage(WORKSPACE_CHANGED)));
+	m->AddItem(new BMenuItem(B_TRANSLATE("Current"), new BMessage(WORKSPACE_CHANGED)));
```
Comment (korli) [inline bepdf/beos/PreferencesWindow.cpp:62]: Translation key shouldn't be changed without a reason.

File: bepdf/Makefile [hunk 7]
```diff
@@ -153,7 +153,8 @@ DEBUGGER :=
 COMPILER_FLAGS = -Wno-write-strings
 
 #	Specify any additional linker flags to be used.
-LINKER_FLAGS = $(shell freetype-config --libs)
+#LINKER_FLAGS = $(shell freetype-config --libs)
```
Comment (korli) [inline bepdf/Makefile:156]:
```suggestion
```

File: bepdf/beos/PreferencesWindow.cpp (hunk 6 above)
Comment (korli) [reply bepdf/beos/PreferencesWindow.cpp:62]:
```suggestion
	m->AddItem(new BMenuItem(B_TRANSLATE("current"), new BMessage(WORKSPACE_CHANGED)));
```

File: xpdf/Makefile [hunk 8]
```diff
@@ -191,10 +191,11 @@ SYMBOLS :=
 DEBUGGER :=
 
 #	Specify any additional compiler flags to be used.
-COMPILER_FLAGS := $(shell freetype-config --cflags)
+COMPILER_FLAGS := $(shell pkg-config --cflags freetype2)
 
 #	Specify any additional linker flags to be used.
-LINKER_FLAGS = $(shell freetype-config --libs)
+#LINKER_FLAGS = $(shell freetype-config --libs)
```
Comment (korli) [inline xpdf/Makefile:197]:
```suggestion
```

**Comment by korli** [comment]
bepdf is already built for x86_gcc2 and x86_64 at HaikuPorts, see https://github.com/haikuports/haikuports/blob/master/haiku-apps/bepdf/bepdf-2.1.4.recipe
This commit is more about modernizing than fixing, correct?

**Pull request merged**